src/mono/mono/metadata/appdomain.c

```diff
@@ -1054,7 +1054,7 @@ mono_domain_owns_vtable_slot (MonoDomain *domain, gpointer vtable_slot)
 	gboolean res;
 
 	mono_domain_lock (domain);
-	res = mono_mempool_contains_addr (domain->mp, vtable_slot);
+	res = mono_mempool_contains_addr (mono_domain_ambient_memory_manager (domain)->mp, vtable_slot);
 	mono_domain_unlock (domain);
```

Comment:
As a general comment, it would be nice to shorten some of these method names/variables names, allocation is very common. Something like 'mm' or 'alloc' or 'allocator'.